app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -19,14 +25,26 @@ open class FeedViewModel : ViewModel() {
     fun getFeedItems(): LiveData<List<FeedItem>> = feedItems
     fun getNetworkErrorEvent(): LiveData<Event<String>> = networkErrorEvent
 
-    init {
+    fun initRepo(feedRepository: FeedRepository) {
+        this.feedRepository = feedRepository
+        feedItems.addSource(feedRepository.getFeedItems(), Observer { feedItems ->
+            isEmpty.value = feedItems.isEmpty()
+            this.feedItems.value = feedItems
+        })
         refresh()
     }
 
+    @SuppressLint("CheckResult")
     fun refresh() {
-        //todo: fix the implementation
-        isLoading.value = false
-        isEmpty.value = true
+        isLoading.value = true
+        feedRepository.refresh().subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({
+                isLoading.value = false
+            }, { error ->
+                isLoading.value = false
+                networkErrorEvent.value = Event(error!!.toString())
```

Comment:
We don't want to crash the app if error happens to be null. Please, provide a default message.
For example: 
`error?.toString() ?: "your default message"`